### PR TITLE
Fix bug preventing selection of unlisted claimant radio button

### PR DESCRIPTION
**Pull request by nanotone**
### Description
This PR fixes a frontend bug in how the Intake Review page handles a new unlisted claimant from the Add Claimant modal.

Previously, the new radio button would not get selected, because of an undefined/null mismatch on the claimant value (participant id). In any case, RadioField complains when its values aren't strings, so I've made sure that the RadioField always gets an empty string instead.

### Testing Plan
1. While performing a Board intake, add an unlisted claimant via the Add Claimant modal.
2. Confirm that new radio button for the unlisted claimant is selected.
3. Confirm that it can be re-selected after selecting a different dependent.

### User Facing Changes
Screenshot of a selectable unlisted claimant on the Review page:

### CODE REVIEW

File: client/app/intake/components/SelectClaimant.jsx [hunk 1]
```diff
@@ -93,12 +93,12 @@ export const SelectClaimant = (props) => {
     setNewClaimant(null);
     setClaimant({ claimant: null, claimantType: 'dependent', claimantNotes: null });
   };
-  const handleSelectDependent = (value) => setClaimant({ claimant: value, claimantType: 'dependent' });
+  const handleSelectDependent = (value) => setClaimant({ claimant: value || null, claimantType: 'dependent' });
```
Comment (nanotone) [inline client/app/intake/components/SelectClaimant.jsx:96]: This function takes a UI event and dispatches a Redux action, so it turns `''` into `null`.

File: client/app/intake/components/SelectClaimant.jsx [hunk 2]
```diff
@@ -93,12 +93,12 @@ export const SelectClaimant = (props) => {
...
   const handleAddClaimant = ({ name, participantId, claimantType, claimantNotes }) => {
     setNewClaimant({
       displayElem: <RemovableRadioLabel
         text={`${name || 'Claimant not listed'}, Attorney`} onRemove={handleRemove} notes={claimantNotes} />,
-      value: participantId,
+      value: participantId ?? '',
```
Comment (nanotone) [inline client/app/intake/components/SelectClaimant.jsx:101]: The claimant's participant ID here, and in the diff below, is used directly by the radio field as one of the possible values, so it gets coerced into a string.

**Comment by jcq** [comment]
> Looks like something isn't working correctly when one switches to a different radio option then comes back to the unlisted one. If you try to submit at that point, the field is marked with an error that it hasn't been filled.
> 

Ah, it's that when you switch to one of the other existing dependents, it sets `claimant` to the participant ID and clears the `claimantType` and `claimantNotes` fields; and when you switch back to the unlisted one, it simply sets `claimant` to `null` and since the other fields remain blank, it's invalid.

**Comment by nanotone** [comment]
@jcq Oh good catch! Somehow I thought I tested that, but clearly not 😅

File: client/app/intake/components/SelectClaimant.jsx [hunk 3]
```diff
@@ -93,14 +93,27 @@ export const SelectClaimant = (props) => {
...
     setClaimant({ claimant: null, claimantType: 'dependent', claimantNotes: null });
   };
-  const handleSelectDependent = (value) => setClaimant({ claimant: value || null, claimantType: 'dependent' });
+  const handleSelectNonVeteran = (value) => {
+    if (newClaimant && value === newClaimant.value) {
+      setClaimant({
+        claimant: value || null,
+        claimantName: newClaimant.claimantName,
+        claimantNotes: newClaimant.claimantNotes,
+        claimantType: newClaimant.claimantType,
```
Comment (nanotone) [inline client/app/intake/components/SelectClaimant.jsx:102]: @jcq: Is there a nifty way to make these three lines snappier?

Comment (jcq) [reply client/app/intake/components/SelectClaimant.jsx:102]: It'd be nice to just use the spread operator, but there are of course more properties in `newClaimant` than you want to pass into here. You could of course destructure to separate the props you don't want from the ones you do with the rest syntax, but that wouldn't gain you a lot of readability or brevity here.

You could perhaps shorten your conditional with the optional chaining operator:
```js
if(value === newClaimant?.value)
```
...but that doesn't gain you a ton either.

**Pull request merged**